pages/guides/frontend-migrations/pf-4-to-pf5.mdx

```diff
@@ -0,0 +1,494 @@
+
+If you see no build errors, you are done with the source code migration.
+
+### The build outputs an about about missing source files!
+
+There is a possibility that some PF source files can't be resolved. The error can look similar to this:
+
+```sh
+ERROR in ./src/components/SourceDetail/ApplicationKebab.js
+Module build failed (from ./node_modules/ts-loader/index.js):
+Error: Unable to find source file for module KebabToggle! The module likely does not have unique file as is included within another file. Please add the entry into the COMPONENTS_CACHE in FEC repository
+    at findComponentModule (/home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:115:15)
+    at /home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:155:33
+    at Array.reduce (<anonymous>)
+    at createDynamicReactCoreImports (/home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:152:30)
+    at visitor (/home/username/path/to/repository/node_modules/@redhat-cloud-services/tsc-transform-imports/index.js:203:24)
+    at visitArrayWorker (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86542:51)
+    at visitNodes2 (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86513:21)
+    at visitLexicalEnvironment (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86569:18)
+    at visitEachChildOfSourceFile (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:87758:13)
+    at Object.visitEachChild (/home/username/path/to/repository/node_modules/typescript/lib/typescript.js:86730:35)
+ @ ./src/components/SourceDetail/ApplicationsCard.js 20:0-50 180:71-87
+ @ ./src/pages/Detail.js 6:0-75 34:36-52
+ @ ./src/Routing.js 8:32-95
+ @ ./src/App.js 7:0-32 33:44-51
+ @ ./src/AppEntry.js 5:0-24 12:24-27
+ @ container entry ./RootApp[0]
+
+```
+
+The patched build checks each patternfly module and tries to resolve their source files. If it fails to find them, it will throw the above error. (The resolution is cached so it will not check one file more than once)
+
+There are two possible reasons for this:
+
+#### The PF migration was not done properly
+
+Some modules no longer exist, were renamed, or were moved to the `@patternfly/react-core/deprecated` directory. Make sure that you have properly migrated your source code and try again.
+
+#### The PF module does not follow naming conventions
+
+Some JS modules do not follow the naming convention so the patched build is unable to retrieve the file. There is a registry of overrides for this case. Please contact the `@crc-experience-services-team` on the `#forum-consoledot-ui` channel and the missing entry will be added.
+
+## Fixing tests
+
+This is probably the longest and most annoying part of the migration. The more tests you have the longer it will most likely take.
```

Comment:
```suggestion
This is probably the longest and most annoying part of the migration. The more tests you have, the longer it will most likely take.
```